src/harness/reference_models/interference/interference.py

```diff
@@ -0,0 +1,669 @@
+   ProtectedEntityType.PPA_AREA : ( PPA_NEIGHBORHOOD_DIST,  PPA_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_CO_CHANNEL : ( FSS_CO_CHANNEL_NEIGHBORHOOD_DIST,  FSS_CO_CHANNEL_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.FSS_BLOCKING : ( FSS_BLOCKING_NEIGHBORHOOD_DIST,  FSS_BLOCKING_NEIGHBORHOOD_DIST),
+   ProtectedEntityType.ESC: (ESC_NEIGHBORHOOD_DIST_A, ESC_NEIGHBORHOOD_DIST_B)
+    }
+
+# Define CBSD grant, i.e., a tuple with named fields of 'latitude',
+# 'longitude', 'height_agl', 'indoor_deployment', 'antenna_azimuth',
+# 'antenna_gain', 'antenna_beamwidth', 'cbsd_category', 
+# 'max_eirp', 'low_frequency', 'high_frequency', 'is_managed_grant'
+CbsdGrantInformation = namedtuple('CbsdGrantInformation',
+                       ['latitude', 'longitude', 'height_agl', 
+                        'indoor_deployment', 'antenna_azimuth', 'antenna_gain',
+                        'antenna_beamwidth', 'cbsd_category', 
+                        'max_eirp', 'low_frequency', 'high_frequency', 
+                        'is_managed_grant'])
+
+# Define protection constraint, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'low_frequency', 'high_frequency'
+ProtectionConstraint = namedtuple('ProtectionConstraint',
+                                  ['latitude', 'longitude', 'low_frequency',
+                                   'high_frequency', 'entity_type'])
+
+# Define FSS Protection Point, i.e., a tuple with named fields of
+# 'latitude', 'longitude', 'height_agl', 'max_gain_dbi', 'pointing_azimuth',
+# 'pointing_elevation' 
+FssProtectionPoint = namedtuple('FssProtectionPoint',
+                                   ['latitude', 'longitude',
+                                    'height_agl', 'max_gain_dbi',
+                                    'pointing_azimuth', 'pointing_elevation'])
+
+
+# Define ESC information, i.e., a tuple with named fields of
+# 'antenna_height', 'antenna_azimuth', 'antenna_gain', 'antenna_pattern_gain'
+EscInformation = namedtuple('EscInformation',
+                                 ['antenna_height', 'antenna_azimuth',
+                                  'antenna_gain', 'antenna_pattern_gain'])
+
+class AggregateInterferenceOutputFormat:
+  """Implementation of output format of aggregate interference 
+
+  This class is used to generate specified output format for post IAP 
+  allowed interference or for aggregate interference to non-DPAs
+  
+  Creates a data structure to store post IAP allowed interference or 
+  aggregate interference to non-DPAs output in the format of 
+  {latitude : {longitude : [ output_first_5MHz_segment, output_first_5MHz_segment ]}}
+  """
+  def __init__(self):
+    self.manager = multiprocessing.Manager()
+    self.aggregate_interference_info = self.manager.dict()
+    self.lock = multiprocessing.Lock()
+
+  def SetAggregateInterferenceInfo(self, latitude, longitude, interference):
+    self.lock.acquire()
+    try:
```

Comment:
usually one do:
```
with self.lock():
   ...
```
The lock acquire/release will be taken care of in a robust way

Reply:
Thanks for the suggestion. Modified as suggested.